Octokit.Tests.Integration/Clients/RepositoryBranchesClientTests.cs

```diff
@@ -543,4 +544,999 @@ public TheDeleteBranchProtectionMethod()
+        {
+            if (_userRepoContext != null)
+                _userRepoContext.Dispose();
+
+            if (_orgRepoContext != null)
+                _orgRepoContext.Dispose();
+        }
+    }
+
+    public class TheAddRequiredStatusChecksContextsMethod : IDisposable
+    {
+        IRepositoryBranchesClient _client;
+        RepositoryContext _userRepoContext;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+
+        public TheAddRequiredStatusChecksContextsMethod()
+        {
+            var github = Helper.GetAuthenticatedClient();
+            _client = github.Repository.Branch;
+
+            _userRepoContext = github.CreateRepositoryWithProtectedBranch().Result;
+            _orgRepoContext = github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task AddRequiredStatusChecksContexts()
+        {
+            var repoOwner = _userRepoContext.RepositoryOwner;
+            var repoName = _userRepoContext.RepositoryName;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.AddRequiredStatusChecksContexts(repoOwner, repoName, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(4, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task AddRequiredStatusChecksContextsWithRepositoryId()
+        {
+            var repoId = _userRepoContext.RepositoryId;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.AddRequiredStatusChecksContexts(repoId, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(4, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task AddRequiredStatusChecksContextsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.AddRequiredStatusChecksContexts(repoOwner, repoName, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(4, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task AddRequiredStatusChecksContextsForOrgRepoWithRepositoryId()
+        {
+            var repoId = _orgRepoContext.RepositoryContext.RepositoryId;
+            var update = new List<string>() { "newMember1", "newMember2" };
```

Comment:
since these are status contexts can we not call them "newMember" but something like "build2" or "deploy" or whatever?